### PR TITLE
Create a bunch of methods using OOP principles that can provide you a…

### PR DESCRIPTION
**Pull request by ArtemHurievskyi**
… list with random figures with random colors

### CODE REVIEW

File: src/main/java/core/basesyntax/Square.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+import java.text.DecimalFormat;
+
+public class Square extends Figure {
+    private double side;
+    private DecimalFormat df = new DecimalFormat("#.00");
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Square.java:7]: bad naming + code duplication. extract it either to some utility class, or to a Figure superclass

File: src/main/java/core/basesyntax/Main.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+            figures[i] = figureSupplier.getRandomFigure();
+        }
+
+        for (int i = (figures.length / 2); i < figures.length; i++) {
+            figures[i] = figureSupplier.getDefaultFigure();
+        }
+
+        for (Figure figure : figures) {
+            System.out.println(figure.toString());
+        }
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Main.java:17]: do it in one for loop

File: src/main/java/core/basesyntax/Main.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    }
+
```
Comment (okuzan) [inline src/main/java/core/basesyntax/Main.java:19]:
```suggestion
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+            case 1:
+                return new Square(randomColor, side1);
+            case 2:
+                return new RightTriangle(randomColor, side1, side2);
+            case 3:
+                return new Rectangle(randomColor, side1, side2);
+            case 4:
+                return new IsoscelesTrapezoid(randomColor, side1, side2, side3);
+            default:
+                return getDefaultFigure();
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:29]:
```suggestion
            case 4:
                return new IsoscelesTrapezoid(randomColor, side1, side2, side3);
            default:
                throw new IllegalStateException("Oops, this shouldn't ever happen! Generated number " + randomFigure);
```
```suggestion
            case 4:
            default:
                return new IsoscelesTrapezoid(randomColor, side1, side2, side3);
```
```suggestion
            default:
                return new IsoscelesTrapezoid(randomColor, side1, side2, side3);
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE.name(), 10);
```
Comment (okuzan) [inline src/main/java/core/basesyntax/FigureSupplier.java:34]: introduce DEFAULT_COLOR, DEFAULT_RADIUS constants

**Review by okuzan — CHANGES_REQUESTED** [review]
Good job overall!

**Review by sarakhmen — APPROVED** [review]
good job :)

**Pull request closed without merging**